### PR TITLE
fix(ml-training): MSE broadcasting bug pred_len=1 in M8 archi (train_{mamba,itransformer,patchtst,tft})

### PR DESCRIPTION
**Pull request by jsboige**
## Summary

Bug detected during walk-forward Mamba launch on GPU 2 (2026-05-11 ai-01):

\`\`\`
UserWarning: Using a target size (torch.Size([64])) that is different to the
input size (torch.Size([64, 1])). This will likely lead to incorrect results
due to broadcasting.
\`\`\`

PyTorch \`MSELoss\` broadcasts \`(B, 1)\` vs \`(B,)\` to \`(B, B)\` matrix → pairwise distances instead of per-sample MSE. Gradient flows in a wrong direction.

Triggers for \`pred_len=1\` across **all 4 M8 architectures**:

| File | Lines (training) | Lines (validation) |
|------|------------------|--------------------|
| train_mamba.py | L487 (FIXED) | L514 (FIXED) |
| train_itransformer.py | L220 (FIXED) | L247 (FIXED) |
| train_patchtst.py | L253 (FIXED) | L280 (FIXED) |
| train_tft.py | L295 (already had guard) | L321 (FIXED) |

## Fix

\`y_batch.view_as(pred)\` to harmonize shapes before \`criterion\` call. Works for any \`pred_len\` (no-op when shapes already match).

## Impact

M8 SOTA sweep verdict (PR #892, **0 BEATS** / 96 runs) is based on training runs that all had this bug for \`pred_len=1\`. The h=5 and h=10 runs may also have the issue if model output is \`[B, pred_len]\` but target loader returns \`[B]\` (TBD).

A walk-forward rerun post-fix is required to validate the verdict robustly (M8 TODO from \`project_m8_verdict_2026-05-10.md\`).

## Test plan

- [x] Mamba walk-forward BTC-USD h=1 relaunched on GPU 2 (BG \`b58fxazq0\`), no warning observed in startup logs (vs initial launch which had the warning).
- [ ] Apple-to-apple comparison with PR #892 results once rerun completes (~5h).
- [ ] If verdict 0 BEATS confirmed post-fix → M8 conclusion robust. If verdict changes → significant finding requiring full sweep redo.

Co-Authored-By: Claude Opus 4.7 (1M context) <noreply@anthropic.com>

### CODE REVIEW

**Review by clusterManager-Myia — APPROVED** [review]
## Review: PR #926 — MSE broadcasting bug pred_len=1 in M8 archi

**Verdict: APPROVE** ✅

### Bug analysis
Correct identification of a real PyTorch broadcasting issue. When `pred_len=1`, model output is `(B, 1)` but the target dataloader returns `(B,)`. PyTorch `MSELoss` broadcasts these to a `(B, B)` matrix, computing pairwise distances instead of per-sample MSE. This silently corrupts gradients.

### Fix quality
- **`y_batch.view_as(pred)`** — minimal, correct, and safe. No-op when shapes already match (e.g. `pred_len>1` where both are `(B, pred_len)`). Applied consistently across all 4 architectures.
- **Consistent pattern**: training + validation loops fixed in mamba, itransformer, patchtst. TFT training (L295) already had the guard — only validation (L321) needed the fix.
- **+11/-7 lines** across 4 files — appropriately scoped.

### Impact awareness ✅
PR body correctly notes that M8 SOTA sweep verdict (PR #892, 0 BEATS / 96 runs) may be affected for `pred_len=1` runs. Rerun plan is documented.

### CI
- ✅ ML Pipeline Tests (CPU): SUCCESS
- ✅ Stale Base Branch: SUCCESS
- ✅ MERGEABLE

No blockers. Ship it.

🤖 Hermes (GLM-5-turbo, cron review-pr)
